core/model/modx/processors/system/menu/remove.class.php

```diff
@@ -16,20 +16,47 @@
  * @package modx
  * @subpackage processors.system.menu
  */
-class modMenuRemoveProcessor extends modObjectRemoveProcessor {
+class modMenuRemoveProcessor extends modObjectRemoveProcessor
+{
     public $classKey = 'modMenu';
-    public $languageTopics = array('action','menu');
+    public $languageTopics = ['action', 'menu'];
     public $permission = 'menus';
     public $objectType = 'menu';
     public $primaryKeyField = 'text';
 
+    /**
+     * @return bool
+     */
+    public function beforeRemove()
+    {
+        $this->removeNested($this->object);
 
-    public function cleanup() {
+        return parent::beforeRemove();
+    }
+
+    /**
+     * @param modMenu $menu
+     */
+    public function removeNested(modMenu $menu)
+    {
+        $criteria = ['parent' => $menu->get('text')];
+
+        if (!$this->modx->getCount($this->classKey, $criteria)) {
+            return;
+        }
+
+        foreach ($this->modx->getIterator($this->classKey, $criteria) as $subMenu) {
+            $this->removeNested($subMenu);
+            $this->modx->runProcessor('system/menu/remove', ['text' => $subMenu->get('text')]);
```

Comment:
Is this `$this->modx->runProcessor` call needed if it's already calling `$this->removeNested()` above?

Reply:
I think `$this->modx->runProcessor` is needed but `$this->removedNested()` isn't because it's being called in `beforeRemove()`.

Reply:
It's needed because of it a recursive call, `removeNested` removes only items that inside, `runProcessor` removes the current item.

Reply:
@JoshuaLuckers Both needed. The first one (in `beforeRemove`) starts removing after that it runs recursively to reach all levels.